### PR TITLE
Improve valueset loading and Elm Parsing

### PR DESCRIPTION
**Pull request by daco101**
Changes:
- only unique valuesets are fetched and added to the measure model
- drc valuesets are shared amongst composite and components
- measure id is removed from version for "Draft" valuesets (not needed with new model)

This PR also improves the Elm Parser to handle large elm files (n^2 to n time complexity for the code that finds the node type), and refactors some of elm parser for clarity and simplicity. I've run some "regressions" with the new vs old ElmParser, the only difference I see is that since I now rely on nokogiri to provide the unescaped strings from the xml (instead of doing manual escaping), "&#xd" / "&#13" are converted to "\r" (which I think is good, since it looks like we needed to replace this with "" in the fronent js code anyways).

Pull requests into cqm-parsers require the following. Submitter and reviewer should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable ("N/A") and :white_check_mark:.

**Submitter:**
- [x] This pull request describes why these changes were made.
- [x] Internal ticket for this PR: https://jira.mitre.org/browse/BONNIE-1935
- [x] Internal ticket links to this PR
- [x] Code diff has been done and been reviewed
- [x] Tests are included and test edge cases
- [x] Tests have been run locally and pass

**Reviewer 1:**

Name: @zlister 
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

**Reviewer 2:**

Name: @jbradl11 
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

### CODE REVIEW

File: lib/measure-loader/elm_parser.rb [hunk 1]
```diff
@@ -2,18 +2,18 @@ module Measures
   class ElmParser
```
Comment (jbradl11) [inline lib/measure-loader/elm_parser.rb]: I don't believe we have ever had tests for this class. This is such an important piece of Bonnie, how difficult would it be to add some?

**Comment by daco101** [comment]
I'm not convinced the "node_type" is actually currently used in the bonnie frontend. It looks like maybe it was used for syntax highlighting in a prior version? I've left it in for now.
